src/openfermion/functionals/get_one_norm.py

```diff
@@ -0,0 +1,142 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Function to calculate the 1-Norm of a molecular Hamiltonian
+(in the restricted formalism) after fermion-to-qubit transformation
+from a MolecularData class. See https://arxiv.org/abs/2103.14753
+for more information on the 1-norm.
+"""
+
+import numpy as np
+
+from openfermion import MolecularData
+
+
+def get_one_norm(mol_or_int, no_constant=None):
+    r"""
+
+    Returns the 1-Norm of a RHF or ROHF Hamiltonian described in
+    https://arxiv.org/abs/2103.14753 after a fermion-to-qubit
+    transformation given nuclear constant, one-body (2D np.array)
+    and two-body (4D np.array) integrals in spatial orbital basis.
+
+    Parameters
+    ----------
+
+    mol_or_int(tuple) : Tuple of (constant, one_body_integrals,
+                                  two_body_integrals)
+    constant(float) : Nuclear repulsion or adjustment to constant shift in
+        Hamiltonian from integrating out core orbitals
+    one_body_integrals(ndarray) : An array of the one-electron integrals having
+        shape of (n_orb, n_orb), where n_orb is the number of spatial orbitals.
+    two_body_integrals(ndarray) : An array of the two-electron integrals having
+        shape of (n_orb, n_orb, n_orb, n_orb).
+
+    -----OR----
+
+    mol_or_int(MolecularData) : MolecularData class object
+
+    -----------
+
+    no_constant (default: None) : If True, do not return the constant term in
+                in the (majorana/qubit) Hamiltonian
+
+    Returns
+    -------
+    one_norm : 1-Norm of the qubit Hamiltonian
+    """
+    if isinstance(mol_or_int, MolecularData):
+        return _get_one_norm_mol(mol_or_int, no_constant=no_constant)
+    else:
+        if no_constant is not None:
```

Comment:
I don't understand this tree - is the idea that no_constant should be either True, False, or None? Also, do we plan the default 'None' to return the constant, or not? I'd recommend setting the default to False or True, and then having a single if statement.

(Also, it seems one of your issues is that you're not covering all three branches of this in your test case).

Reply:
Hmm that is what it was before. I thought this is what Nick meant with "Use default None and allow the user to specify", but I'm probably wrong.

Reply:
Heh, fair enough. @ncrubin , can you comment here? I think that a 'None' default is less explanatory as it doesn't tell the user whether the constant will be by default returned or not. Is there any issue with using True/False as a default?

Reply:
Hi @Emieeel , just chatted with @ncrubin about this, and I think that the original comment was misunderstood. The point is that instead of taking a tuple as input with all the different data, you should split it up into the different parts, i.e.

def get_one_norm_int(constant, one_body_integrals, two_body_integrals, no_constant=True),

and make a second function 'get_one_norm_mol' which takes a molecular Hamiltonian.

Sorry for the delay in sorting this out!

Reply:
Could also do a 'get_one_norm' and 'get_one_norm_mol' as a convenience function.

Reply:
I'm sorry it took so long, I forgot a bit about it; Is it fine now? I defined two functions: get_one_norm_int and get_one_norm_mol, with no_constant=True.